### DISCUSSION TITLE
writer.fillTextbox and show_pdf_page conflict in some  pdf

### DISCUSSION THREAD

**Discussion by HelloKitty2022** [pymupdf/PyMuPDF#932] [Looking for help] [answered]
rawfile=  r"C:\1.pdf"
tmpfile=r"C:\1bak.pdf"
shutil.copy(rawfile,tmpfile)
srcdoc=fitz.open(rawfile)
tmpdoc=fitz.open(tmpfile)
outdoc=fitz.open()
for pagei in tmpdoc:
    width=pagei.rect.width
    height=pagei.rect.height
    blocks=pagei.get_text("blocks")
    for b in blocks:
        text=b[4]
        writer = fitz.TextWriter(pagei.rect, color=blue)  # start a text writer
        writer.fillTextbox(  # fill in above text
            fitz.Rect(10,10,100,100),  # keep text inside this
            text,  # the text
            align=fitz.TEXT_ALIGN_LEFT,  # alignment
            warn=True,  # keep going if too much text
            fontsize=8,
            font=font,
        )
        writer.writeText(pagei)
    outpage = outdoc.new_page(-1,
            width = width*2,
            height = height)
    outpage.show_pdf_page(pagei.rect,
                    srcdoc, # input document
                    pagei.number) # input page number
    outpage.show_pdf_page(fitz.Rect(pagei.rect.tr,outpage.rect.br),
                tmpdoc, # input document
                pagei.number) # input page number
srcdoc.close()
tmpdoc.close()
outdoc.save("a.pdf",garbage=3, deflate=True)
thereis some thing wrong 
![微信截图_20210305040059](https://user-images.githubusercontent.com/30066186/110022858-6daba100-7d67-11eb-80ca-d45348ec8e76.png)
1. i found  if the  writer.fillTextbox's text="123", that works .
2. only occured of  some special  pdf   but i dont konw the reason
[1.pdf](https://github.com/pymupdf/PyMuPDF/files/6086358/1.pdf)

**Comment by JorjMcKie** [comment]
I don't understand what the problem is. Can you please describe what you are expecting and what is happening instead?
Your program code is very difficult to read - all indentations are missing. Please send the script attached, or at least display it as code, not as text.

**Comment by HelloKitty2022** [comment]
sorry about that ，

i extract text from a.pdf , writer the translated text to the page , 
after that combine the raw pdf page with the writed page  into one page
but occured erro as the pic 



[test.zip](https://github.com/pymupdf/PyMuPDF/files/6089136/test.zip)

**Comment by JorjMcKie** [comment]
Thanks for the material. Let me put this under discussions, because there is no PyMuPDF issue.

**Comment by JorjMcKie** [comment]
There are still things I don't understand.
This is what I **_did_** understand:
* you want to copy a PDF to an output file, that has pages with the double of the input width
* the left part of each output page should contain the input page, the right part should contain the input again, but with translated text

If we agree on this, here are my comments:
1. You do not need to copy the input file: you can show each input page two or more times on the output!
2. When inserting text, you should do this on the **_output_** page - do **_not modify the input_** in the middle of the copy process. This has caused the error.
3. Your text insertions always go to the same rectangle - for the text of all the input text blocks.
4. When extracting text only (no images), best use flags=0: this will prevent images to be extracted, and thus improve performance and save you selecting text blocks yourself.
5. If you want to insert translated text on the right side of the output pag (presumably Chinese), then I would cover the ld English text.
6. Also, using `fill_textbox()` may be a bad idea, because it may contain line breaks that you do not expect. Use directly positioned text from the input instead (i.e. `append()` method of text writer).

I will send you a suggestion in next post.

**Comment by JorjMcKie** [comment, marked as answer]
[issue-931.zip](https://github.com/pymupdf/PyMuPDF/files/6090105/issue-931.zip)

**Reply by HelloKitty2022** [reply]
its exactly what  i means 
so grateful for your code and suggestion，especially for amateur coder ，its means a lot
